cli/mender-artifact/copy.go

```diff
@@ -96,10 +100,7 @@ func Copy(c *cli.Context) (err error) {
 	}
 
 	_, err = io.Copy(w, r)
-	// Ignore the close error in case of an io.Copy error.
-	if cerr := w.Close(); cerr != nil && err != nil {
-		err = cerr
-	}
+
```

Comment:
Just wondering.

So using defer w.Close will hide any error that could occur while actually closing, right? Is this a good practice? It seems like it is something we do a lot...

But for all these, can we instead do something like the following?

```
  defer func() {
    if ferr := f.Close(); ferr != nil {
      err = ferr
    }
  }()
```

Reply:
Yes, I think you are right here. This could potentially be misleading, as an error repacking, say an artifact, in the `Close()` method would indeed be camouflaged by this. Let me fix it up. Good eye!

Reply:
Keep in mind that this problem was already present in the old code. And it is probably present in other source files as well.

IMO, if you decide to fix it, it should be done everywhere to avoid inconsistency in the design. So please have a look around for suspicious `.close()` calls.

Reply:
Yes, of course :)

Reply:
@lluiscampos Now checks the error message of all the writers where it is necessary. Feel free to have a look.